### PR TITLE
core: get_dataset_lock_by_rse unproper handled exception #941

### PR DESCRIPTION
**Pull request by jwackito**
core: get_dataset_lock_by_rse unproper handled exception #941

### CODE REVIEW

File: lib/rucio/web/rest/webpy/v1/lock.py [hunk 1]
```diff
@@ -64,6 +64,8 @@ def GET(self, rse):
                     yield render_json(**lock) + '\n'
             else:
                 raise InternalError('Wrong did_type specified')
+        except RSENotFound as error:
+            raise generate_http_error(404, 'RSENotFound', error.args[0])
```
Comment (mlassnig) [inline lib/rucio/web/rest/webpy/v1/lock.py]: ```error.__class__.__name__```

?

Comment (jwackito) [reply lib/rucio/web/rest/webpy/v1/lock.py]: /scope.py:            raise generate_http_error(404, 'ScopeNotFound', 'no scopes found for account ID \'%s\'' % account)
lib/rucio/web/rest/webpy/v1/subscription.py:from rucio.common.utils import generate_http_error, APIEncoder, render_json
lib/rucio/web/rest/webpy/v1/subscription.py:            raise generate_http_error(404, 'SubscriptionNotFound', error.args[0])
lib/rucio/web/rest/webpy/v1/subscription.py:            raise generate_http_error(400, 'ValueError', 'Cannot decode json parameter list')
lib/rucio/web/rest/webpy/v1/subscription.py:            raise generate_http_error(400, 'InvalidObject', error.args[0])
lib/rucio/web/rest/webpy/v1/subscription.py:            raise generate_http_error(401, 'AccessDenied', error.args[0])
lib/rucio/web/rest/webpy/v1/subscription.py:            raise generate_http_error(404, 'SubscriptionNotFound', error.args[0])
lib/rucio/web/rest/webpy/v1/subscription.py:            raise generate_http_error(500, error.__class__.__name__, error.args[0])
lib/rucio/web/rest/webpy/v1/subscription.py:            raise generate_http_error(400, 'ValueError', 'Cannot decode json parameter list')
lib/rucio/web/rest/webpy/v1/subscription.py:            raise generate_http_error(400, 'InvalidObject', error.args[0])
lib/rucio/web/rest/webpy/v1/subscription.py:            raise generate_http_error(401, 'AccessDenied', error.args[0])
lib/rucio/web/rest/webpy/v1/subscription.py:            raise generate_http_error(409, 'SubscriptionDuplicate', error.args[0])
lib/rucio/web/rest/webpy/v1/subscription.py:            raise generate_http_error(500, error.__class__.__name__, error.args[0])
lib/rucio/web/rest/webpy/v1/subscription.py:            raise generate_http_error(404, 'RuleNotFound', error.args[0])
lib/rucio/web/rest/webpy/v1/subscription.py:            raise generate_http_error(404, 'SubscriptionNotFound', error[0])
lib/rucio/web/rest/webpy/v1/subscription.py:            raise generate_http_error(500, error.__class__.__name__, error.args[0])
lib/rucio/web/rest/webpy/v1/subscription.py:            raise generate_http_error(500, error.__class__.__name__, error.args[0])
lib/rucio/web/rest/webpy/v1/subscription.py:            raise generate_http_error(404, 'SubscriptionNotFound', error.args[0])
lib/rucio/web/rest/webpy/v1/subscription.py:            raise generate_http_error(500, error.__class__.__name__, error.args[0])
lib/rucio/web/rest/webpy/v1/temporary_did.py:from rucio.common.utils import generate_http_error
lib/rucio/web/rest/webpy/v1/temporary_did.py:            raise generate_http_error(400, 'ValueError', 'Cannot decode json parameter list')
lib/rucio/web/rest/webpy/v1/temporary_did.py:            raise generate_http_error(500, error.__class__.__name__, error.args[0])
lib/rucio/web/rest/webpy/v1/trace.py:from rucio.common.utils import generate_http_error
lib/rucio/web/rest/webpy/v1/trace.py:            raise generate_http_error(400, 'ValueError', 'Cannot decode json parameter list')
lib/rucio/web/ui/main.py:from rucio.common.utils import generate_http_error
lib/rucio/web/ui/main.py:            raise generate_http_error(401, 'CannotAuthenticate', 'Cannot get token')
lib/rucio/web/ui/main.py:            raise generate_http_error(503, str(type(err)), str(err))

Comment (jwackito) [reply lib/rucio/web/rest/webpy/v1/lock.py]: should I use error.__class__.__name__.?

Comment (mlassnig) [reply lib/rucio/web/rest/webpy/v1/lock.py]: hmmmm.. at least in webpy all of them should use ```error.__class__.__name__```
(you can leave out the webui for now)

can you also check that the same thing is correct in web/rest/flask?

Comment (jwackito) [reply lib/rucio/web/rest/webpy/v1/lock.py]: is not... I will amend it and push --force

Comment (jwackito) [reply lib/rucio/web/rest/webpy/v1/lock.py]: done...

Comment (mlassnig) [reply lib/rucio/web/rest/webpy/v1/lock.py]: lib/rucio/web/rest/lock.py:67:16: F821 undefined name 'RSENotFound'
lib/rucio/web/rest/flaskapi/v1/lock.py:61:16: F821 undefined name 'RSENotFound'
lib/rucio/web/rest/flaskapi/v1/lock.py:62:19: F821 undefined name 'generate_http_error'
lib/rucio/web/rest/webpy/v1/lock.py:67:16: F821 undefined name 'RSENotFound'

last few problems

**Review by bari12 — APPROVED** [review]

**Pull request merged**